patches/minecraft/net/minecraft/block/Block.java.patch

```diff
@@ -1354,6 +1354,20 @@
 +        return func_185467_w();
 +    }
 +
++    /**
++     * Gets color this block adds to a beacon beam
++     *
++     * Return a float array with a red, green and blue element or null for no change
++     *
++     * @param state The state
++     * @param world The world
++     * @return The color or null
++     */
++    public float[] getBeaconColor(IBlockState state, World world)
```

Comment:
most blocks will need to know where in the world they are to do stuff appropriately